integration-tests/src/test/groovy/com/okta/sdk/tests/it/CrudTestSupport.groovy

```diff
@@ -33,69 +36,74 @@ trait CrudTestSupport implements ClientProvider {
     void basicCrudTest() {
 
         Client client = getClient()
-
         preTestSetup(client)
 
-        // get the size of initial list
-        int preCount = getCount(client)
-        slowItDown()
-
         // Create a resource
         def resource = create(client)
-        slowItDown()
-
-        // count the resource after creating
-        int count = getCount(client)
-        assertThat "More then one resource was created", count, is(preCount + 1)
-        slowItDown()
 
         // getting the resource again should result in the same object
-        assertThat read(client, resource.id), equalTo(resource)
+        def readResource = read(client, resource.id)
+        assertThat readResource, notNullValue()
+        assertThat readResource, equalTo(resource)
 
         // update the resource
         update(client, resource)
-        slowItDown()
-
-        count = getCount(client)
-        assertThat "New resource was created while attempting to update", count, is(preCount + 1)
-        slowItDown()
+        assertUpdate(client, read(client, resource.id))
 
         // delete the resource
         delete(client, resource)
-        slowItDown()
-
-        count = getCount(client)
-        assertThat "Resource was not deleted", count, is(preCount)
-        slowItDown()
+        assertDelete(client, resource)
     }
 
-    void slowItDown() {
-        Thread.sleep(200)
-    }
+    @Test
+    void basicListTest() {
 
-    void preTestSetup(Client client) {
+        // Create a resource
+        def resource = create(client)
+        // search the resource collection looking for the new resource
+        Optional optional = getResourceListStream(client)
+                                .filter {it.id == resource.id}
+                                .findFirst()
 
+        // make sure it exists
+        assertThat "New resource with id ${resource.id} was not found in list resource.", optional.isPresent()
     }
 
+    void preTestSetup(Client client) {}
+
     // create
```

Comment:
Not sure if we need these redundant comments

Reply:
+1